### ISSUE TITLE
spinner no longer working

### ISSUE THREAD

**Issue by ashleygwilliams** [ashleygwilliams/wasm-pack#145]
the spinner used to spin- it seems to not now. this isn't a huge problem, but it'd be nice to fix by the next release!

**Comment by mgattozzi** [comment]
I feel like this might have been introduced with the PBAR module stuff I did. Possibly related to how the multibar works. Would need to look into it. Might also just need a package update, but I haven't looked into the underlying reason why yet.

**Comment by migerh** [comment]
I had a look at this and it is indeed related to the MultiProgressBar. When a progress bar is added to a multibar, the drawing target is changed from the default `stderr` to a remote draw target, whatever that is. See also the [multibar docs](https://docs.rs/indicatif/0.9.0/indicatif/struct.MultiProgress.html#methods), especially the `add` method. I don't know how the multibar is supposed to intercept its progress bar draws to the remote target, but it looks like it doesn't work for spinners.

Before I dive any deeper into multibars and remote targets: Is there a need for multiple spinners to be in use at the same time? Right now it looks like there's only one spinner spinning at any given time so it might make sense to ditch the multibar and instead just return a new progress from `PBAR.message()`. I have a patch that does exactly that and it seems to work just fine.

**Comment by migerh** [comment]
Turns out there are multiple spinners instantiated at the same time. All those warnings and infos and errors are spinners as well but they don't spin because they immediately `finish()`. These wannabe spinners interfere with the real spinners so that a real spinner will overwrite the last warning/error/info. So with my patch it looks like this

```
  [4/7] Writing a package.json...
  :-) [WARN]: Field description is missing from Cargo.toml. It is not necessary, but recommended
  :-) [WARN]: Field repository is missing from Cargo.toml. It is not necessary, but recommended
  [4/7] Writing a package.json...
  [5/7] Copying over your README...
```

instead of this

```
| [4/7] Writing a package.json...
| :-) [WARN]: Field description is missing from Cargo.toml. It is not necessary, but recommended
| :-) [WARN]: Field repository is missing from Cargo.toml. It is not necessary, but recommended
| :-) [WARN]: Field license is missing from Cargo.toml. It is not necessary, but recommended
| [5/7] Copying over your README...
```

I'll keep looking…

**Comment by migerh** [comment]
Small update: I think the issue is with the indicatif crate. I opened an issue [here](https://github.com/mitsuhiko/indicatif/issues/52).

Short summary: Spinners in a multibar do not spin while a child process spawned with `std::process::Command` runs.

**Comment by ashleygwilliams** [comment]
fixed by #180
